### ISSUE TITLE
Iterating over features?

### ISSUE THREAD

**Issue by tkell** [algorithmic-music-exploration/amen#38]
One of my favorite things about Remix was the ability to do `for beat in beats`, and so on.  

We don't currently have an iterator over features, e.g. `for amp in amplitudes`.

Likewise, we don't have an easy way to get TimeSlices and the features to use to manipulate them, unless we do something like:
```
amps = audio.features['amplitude'].at(audio.timings['beats'])
for feature, beat in zip(amps, audio.timings['beats']):
    # do things to each beat based on feature
```

I feel like we should: 
a) make the data in the dataframe of a feature iterable.
b) Allow a feature to reference its timings.  Something like `feature.with_time()`, maybe?

To contrast:
```
amps = audio.features['amplitude'].at(audio.timings['beats'])
for feature, beat in amps.with_time()
    # do things to each beat based on feature
```
A problem with that is that feature objects that have not been resampled do not have any TimeSlices to reference. 

Thoughts?

**Comment by bmcfee** [comment]
Option (b) is similar in spirit to `dict.iteritems()` which generates key-value pairs from a dict.  I think something similar could be done here, maybe like:

```python
amp = audio.features['amplitude']
for beat, feature in amp.iter_at(audio.timings['beats']):
    # do things to each beat based on feature
```

I'd generally put the timing info first, for congruence with `iteritems`.

**Comment by tkell** [comment]
Timing info first, can do.

I feel like this is a function that should be called by a Feature that has already been indexed, rather than making a new way of indexing a Feature.  

This also allows us to put what timing a Feature has been indexed by in __repr__, which seems nice.  

I did some stuff around part a) in #39, too.

**Comment by tkell** [comment]
Update:  Got a sketch of `with_time()` in #39

**Comment by tkell** [comment]
Added with_time for `FeatureCollection`, also in #39

**Comment by tkell** [comment]
And merged & closed #39.  Please add other issues with it as needed!
